WPFHospitalEditor/Equipment.xaml.cs

```diff
@@ -0,0 +1,151 @@
+﻿using HealthcareBase.Dto;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for Equipment.xaml
+    /// </summary>
+    public partial class Equipment : Window
+    {
+        Dictionary<String,EquipmentDto> allEquipment = new Dictionary<string, EquipmentDto>();
+        private List<string> labelContent = new List<string>();
+        private List<string> value = new List<string>();
+        private List<Label> labels = new List<Label>();
+        private List<TextBox> textBoxes = new List<TextBox>();
+
+        public Equipment(Dictionary<String, EquipmentDto> allEquipment)
+        {
+            InitializeComponent();
+            this.allEquipment = allEquipment;
+            this.Height = (allEquipment.Count() + 2) * 50 + 30;
+            ModifyDynamicWPFGrid();
+        }
+
+        private void ModifyDynamicWPFGrid()
+        {
+            createColumns();
+            createRows(allEquipment.Count());
+            createRowContent(allEquipment.Count());
+            Border.Child = DynamicGrid;
+        }
+
+        private void createColumns()
+        {
+            createOneColumn(15);
+            createOneColumn(150);
+            createOneColumn(150);
+            createOneColumn(20);
+        }
+
+        private void createOneColumn(int width)
+        {
+            ColumnDefinition borderColumn1 = new ColumnDefinition();
+            borderColumn1.Width = new GridLength(width);
+            DynamicGrid.ColumnDefinitions.Add(borderColumn1);
+        }
+
+        private void createRows(int rowNumbers)
+        {
+            createOneRow(2);
+            for (int i = 0; i <= rowNumbers + 1; i++)
+            {
+                createOneRow(50);
+            }
+        }
+
+        private void createOneRow(int height)
+        {
+            RowDefinition gridRow1 = new RowDefinition();
+            gridRow1.Height = new GridLength(height);
+            DynamicGrid.RowDefinitions.Add(gridRow1);
+        }
+
+        private void createRowContent(int rowNumbers)
+        {
+                for (int i = 0; i < rowNumbers; i++)
+                {
+                    if (allEquipment.Count() == 0) break;
+                    setRowContent(i);
+                }
```

Comment:
This if will never trigger because allEquipment.Count() is parameter in this method.

Reply:
That wasn't even needed, i deleted it. The check was made before opening EquipmentWindow